tensorboard/webapp/widgets/data_table/data_table_component.ts

```diff
@@ -200,4 +221,61 @@ export class DataTableComponent {
     }
     this.sortDataBy.emit({header, order: SortingOrder.ASCENDING});
   }
+
+  dragStart(header: ColumnHeaders) {
+    this.draggingHeader = header;
+
+    // This stop the end drag animation
+    document.addEventListener('dragover', preventDefault);
+  }
+
+  dragEnd() {
+    if (!this.draggingHeader || !this.highlightedColumn) {
+      return;
+    }
+
+    this.orderColumns.emit(
+      this.moveHeader(
+        this.headers.indexOf(this.draggingHeader),
+        this.headers.indexOf(this.highlightedColumn)
+      )
+    );
+    this.draggingHeader = undefined;
+    this.highlightedColumn = undefined;
+    document.removeEventListener('dragover', preventDefault);
+  }
+
+  dragEnter(header: ColumnHeaders) {
+    if (!this.draggingHeader) {
+      return;
+    }
+    if (
+      this.headers.indexOf(header) < this.headers.indexOf(this.draggingHeader)
+    ) {
+      this.highlightSide = Side.LEFT;
+    } else {
+      this.highlightSide = Side.RIGHT;
+    }
+    this.highlightedColumn = header;
+  }
+
+  // Move the item at sourceIndex to destinationIndex
+  moveHeader(sourceIndex: number, destinationIndex: number) {
+    const newHeaders = [...this.headers];
+    // Delete from original location
+    newHeaders.splice(sourceIndex, 1);
+    // Insert at destinationIndex.
+    newHeaders.splice(destinationIndex, 0, this.headers[sourceIndex]);
+    return newHeaders;
+  }
+
+  getHeaderHighlightStyle(header: ColumnHeaders) {
+    return {
+      highlight: header === this.highlightedColumn,
+      'highlight-border-right':
+        header === this.highlightedColumn && this.highlightSide === Side.RIGHT,
+      'highlight-border-left':
+        header === this.highlightedColumn && this.highlightSide === Side.LEFT,
+    };
+  }
```

Comment:
nitpick (optional): all of these classes seem to be gated by the same check, how do you feel about a guard clause instead of repeated enumeration?
```suggestion
  getHeaderHighlightStyle(header: ColumnHeaders) {
    if (header !== this.highlightedColumn) {
      return {};
    }
    return {
      highlight: true,
      'highlight-border-right': this.highlightSide === Side.RIGHT,
      'highlight-border-left': this.highlightSide === Side.LEFT,
    };
  }
```

Reply:
Yea that is cleaner and more readable for sure. Changed.